### ISSUE TITLE
Model forward pass outputs are identical for all inputs, but only when batch size exceeds some large value

### ISSUE THREAD

**Issue by austinhoag** [keras-team/tf-keras#361]
**System information**.
TensorFlow installed using miniconda, from this YAML file: https://github.com/jeffheaton/t81_558_deep_learning/blob/master/tensorflow-apple-metal-conda.yml

Platform: macOS-13.0-arm64-arm-64bit
Tensor Flow Version: 2.10.0
Keras Version: 2.10.0
Python 3.9.13

GPU model and memory: Apple M1 8 GB

**Describe the problem**.
 
I am running a simple convnet on MNIST. There are 10 output classes and the model uses a softmax activation in its output so the final outputs are probabilities for each of the 10 classes. I have found that when I do a manual forward pass (on a trained on randomly initialized model), the output is the same for each corresponding training sample. However, this only happens when the number of training samples I use in the forward pass exceeds some value (in my case around 40,000). I have looked at hundreds of other issues on here using the keyword search and none of them has addressed this problem.

**Describe the current behavior**.
Minimal working example. This is an untrained model initialized using the default initialization. I have found that the weights I set in the model don't change the behavior I am seeing. For example, if I train the model first and then evaluate it, I see this behavior.
```python
import tensorflow as tf
import tensorflow.keras

# Get the data
mnist = tf.keras.datasets.mnist
(x_train, y_train), (x_test, y_test) = mnist.load_data()

# Reshape and normalize the data
x_train=x_train.reshape(x_train.shape[0], x_train.shape[1], x_train.shape[2], 1)
x_train=(x_train / 255.0).astype('float32')

x_train.shape # prints (60000,28,28,1)

# Define the model and build it
num_classes = 10
input_shape = (28, 28, 1)
model = tf.keras.models.Sequential([
    tf.keras.layers.Conv2D(32, (5,5), padding='same', activation='relu', input_shape=input_shape),
    tf.keras.layers.Conv2D(32, (5,5), padding='same', activation='relu'),
    tf.keras.layers.MaxPool2D(),
    tf.keras.layers.Dropout(0.25),
    tf.keras.layers.Conv2D(64, (3,3), padding='same', activation='relu'),
    tf.keras.layers.Conv2D(64, (3,3), padding='same', activation='relu'),
    tf.keras.layers.MaxPool2D(strides=(2,2)),
    tf.keras.layers.Dropout(0.25),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dropout(0.5),
    tf.keras.layers.Dense(num_classes, activation='softmax')
])
model.build(input_shape=input_shape)
```
When I run the above code I get the warning:
```python
Metal device set to: Apple M1
2022-11-17 10:03:11.366801: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:306] Could not identify NUMA node of platform GPU ID 0, defaulting to 0. Your kernel may not have been built with NUMA support.
2022-11-17 10:03:11.367148: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:272] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 0 MB memory) -> physical PluggableDevice (device: 0, name: METAL, pci bus id: <undefined>)
```
No idea what `NUMA` is, but perhaps this is relevant to the issue I am seeing. 

## Do a forward pass with a small batch (size=64 training samples)
```python
batch_size=64
x_batch = x_train[0:batch_size]
x_batch.shape # prints (64,28,28,1) 

preds_batch1 = model(x_batch)
print(preds_batch1)
```
Which prints:
```python
<tf.Tensor: shape=(64, 10), dtype=float32, numpy=
array([[0.09595986, 0.1002916 , 0.10184481, 0.09941996, 0.1021289 ,
        0.10538033, 0.0985291 , 0.09900756, 0.09954101, 0.09789699],
       [0.09660684, 0.10389937, 0.10340614, 0.09841089, 0.10144418,
        0.1034528 , 0.10085803, 0.09899762, 0.09854244, 0.09438165],
       [0.09537848, 0.10107057, 0.10281242, 0.09908805, 0.10202667,
        0.10022601, 0.10605689, 0.0969203 , 0.09741242, 0.09900815],
       [0.09672208, 0.10293625, 0.10405749, 0.10432041, 0.09796295,
        0.10368358, 0.09960631, 0.09782498, 0.09770255, 0.09518337],
       [0.09558774, 0.10257296, 0.10145756, 0.10067602, 0.10090942,
        0.10331049, 0.10201143, 0.09675287, 0.09864451, 0.09807698],
       [0.09662391, 0.10230789, 0.10366919, 0.09903345, 0.09934687,
...
```
The outputs (each vector) are different for each of the 64 inputs, which is what I'd expect. No issues here.

## Do a forward pass with a big batch (size=40000 training samples)
```python
batch_size=40000
x_batch = x_train[0:batch_size]
preds_batch2 = model(x_batch)
print(preds_batch2)
```
Which prints:
```python
<tf.Tensor: shape=(40000, 10), dtype=float32, numpy=
array([[0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1],
       [0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1],
       [0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1],
       ...,
       [0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1],
       [0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1],
       [0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1]], dtype=float32)>
```
All of the output vectors are the same and they predict equal probability `0.1 = 1/10` for each class.

I have verified that the samples in `x_batch` are different. You can verify for yourself by reproducing this example.

You might ask why I would pass such a large batch in a single forward pass. I'm doing this when evaluating the model, not training. It turns out in my application I cannot use smaller batches for the evaluation step. Even if I could batch my data for evaluation, I still see this behavior as a problem. For example, how small would I need to make my batches to ensure that I won't see this unwanted behavior? But what I really want to know is: why is this happening in the first place? 

And yes, I can use a smaller test set, but I am building software that others will use. They could use a larger test set, and given that the above unwanted behavior does not raise an error (on its own), I want to understand it so I can properly safeguard against it.
